mozci/task.py

```diff
@@ -285,6 +287,31 @@ def _should_retrigger(self, task):
 
         return task["tags"].get("retrigger", "false")
 
+    def backfill(self, push):
+        """This function implements ability to perform backfills on tasks"""
+        decision_task = push.decision_task
+        actions = decision_task.get_artifact("public/actions.json")
+        backfill_action = next(
+            action for action in actions["actions"] if action["name"] == "backfill"
+        )
+        assert backfill_action["kind"] == "hook"
+
+        hook_payload = jsone.render(
+            backfill_action["hookPayload"],
+            context={
+                "taskId": self.id,
+                "taskGroupId": decision_task.id,
+                "input": {"times": 5 if self.classification == "intermittent" else 1},
+            },
+        )
+
+        logger.info("Backfilling task '{}'".format(self.tags.get("label", "")))
+        hooks = taskcluster.Hooks(get_taskcluster_options())
```

Comment:
The hook is in firefox-ci rather than community-tc, with this you'd be using the second

Reply:
We have an example (exactly opposite) where we have a program running in firefox-ci that wants to trigger a hook on community-tc.
In that case, we define a `community_taskcluster_config` at https://github.com/mozilla/code-review/blob/2a8437bb3c34205c5096d113080775baa13a7cf9/events/code_review_events/__init__.py#L5, we authenticate at https://github.com/mozilla/code-review/blob/2a8437bb3c34205c5096d113080775baa13a7cf9/events/code_review_events/cli.py#L63-L67 with the client ID and access token we have defined in the secrets, and then we get the "hooks" service at https://github.com/mozilla/code-review/blob/2a8437bb3c34205c5096d113080775baa13a7cf9/events/code_review_events/bugbug_utils.py#L73.

We have to do something similar here.

Reply:
Oh okay, I didn't pay attention to that, thank you for the heads up!

Reply:
I think it won't work because from community-tc we wouldn't have the permissions to trigger the hook.
Probably that code snippet was written with the idea that we'd be running from firefox-ci.

Reply:
I think it would work if you also add
```Python
options["credentials"] = {
    "clientId": client_id,
    "accessToken": access_token,
}
```
to the options

Reply:
client_id and access_token will have to be defined as secrets